baseband/mark4/base.py

```diff
@@ -23,35 +23,41 @@
 class Mark4FileReader(VLBIFileBase):
     """Simple reader for Mark 4 files.
 
-    Adds ``read_frame`` and ``find_frame`` methods to the VLBI file wrapper.
+    Adds ``read_frame`` and ``locate_frame`` methods to the VLBI file wrapper.
+
+    Parameters
+    ----------
+    decade : int or None
+        Decade in which the observations were taken.  Can instead pass an
+        approximate `ref_time`.
+    ref_time : `~astropy.time.Time` or None
+        Reference time within 4 years of the observation time.  Used only
+        if `decade` is not given.
+    ntrack : int or None, optional.
+        Number of Mark 4 bitstreams.  Can be determined automatically as
+        part of locating the first frame.
     """
+    def __init__(self, fh_raw, decade=None, ref_time=None, ntrack=None):
```

Comment:
As with the stream reader, `ntrack` should come before `decade` and `ref_time`.

Though, the original reason why I had that order was for back compatibility with `read_frame(ntrack, decade)`.  Now that we've changed `read_frame`, perhaps it makes more sense to have `sample_rate, decade, ref_time, ntrack, squeeze...` across Mark 4 classes?

Reply:
Ahh, you did change it in the stream reader.  In that case, the order should change across stream classes too.